docs/testplan/srv6/SRv6-phoenixwing-traffic-testplan-using-trex.md

```diff
@@ -0,0 +1,416 @@
+```
+
+
+#### how to check packet encapsulation
+PTF provides a complete set of packet inspection capabilities. Packet sent by Trex, is transmitted in topo links, and can all be captured by the PTF framework. We will reuse PTF's packet verification capabilities to check packet encapsulation, However we can do some simple encapsulation on the ptf functions, to make them more easy-to-use for SRv6 traffic. 
+
+3 APIs are provided to:
+ 1) check raw IP(v6) packet, 2) check IP(v6)inIPv6 packet (VPN case), 3) IP(v6)inIPv6withSRH packet (TE case).
+
+```
+ptf_port_for_backplane = 18
+ptf_port_for_P2_to_P1 = 16
+ptf_port_for_P2_to_P3 = 36
+ptf_port_for_P4_to_P1 = 17
+ptf_port_for_P4_to_P3 = 37
+ptf_port_for_PE3_to_P2 = 39
+ptf_port_for_PE3_to_P4 = 40
+ptf_port_for_P1_to_PE1 = 28
+ptf_port_for_P1_to_PE2 = 29
+ptf_port_for_P3_to_PE1 = 34
+ptf_port_for_P3_to_PE2 = 35
+
+#check raw IP/IPv6 packets to CE, packet should be in format as:
+# Ether()/IP(dst = dst_ip, tos=(dscp<<2))/UDP()
+check_topo_recv_pkt_raw(ptfadapter, port=0, dst_ip="", dscp = 0, no_packet = False)
+#check SRv6 VPN packets(IPinIPv6 or IPv6inIPv6), packet should be in format as:
+# Ether()/IPv6(dst=vpnsid, nh=4)/IP(dst = dst_ip, tos=(dscp<<2))/UDP()
+check_topo_recv_pkt_vpn(ptfadapter, port=0, dst_ip="", dscp = 0, vpnsid = "", no_packet = False)
+#check SRv6 TE packets(IPinIPv6Srh or IPv6inIPv6Srh), packet should be in format as:
+# Ether()/IPv6(dst=segment, nh=43)/IPv6ExtHdrSegmentRouting(addresses=[vpnsid], nh=4, segleft=1)/IP(dst = dst_ip, tos=(dscp<<2))/UDP()
+check_topo_recv_pkt_srh_te(ptfadapter, port=0, dst_ip="", dscp = 0, vpnsid = "", segment = "", no_packet = False)
+```
+
+#### test case example
+A typical test case example, which demonstrates how to use the Trex functions and PTF functions to check the SRv6 traffic.
+
+```
+    test_ipv4_dip = "192.168.0.1"
+    result = trex_run(test_ipv4_dip, dscp = 4, duration = 10)
+    check_list = {"ptf_tot_rx": 10000, "ptf_tot_tx": 10000, "P1_tx_to_PE1": 2500, "P1_tx_to_PE2": 2500, "P2_tx_to_PE1": 2500, "P2_tx_to_PE2": 2500}
+    pytest_assert(thresh_check(result, check_list))
+
+    check_topo_recv_pkt_raw(ptfadapter, port=ptf_port_for_backplane, dst_ip=test_ipv4_dip, dscp = 4)
+    check_topo_recv_pkt_vpn(ptfadapter, port=ptf_port_for_P1_to_PE1, dst_ip=test_ipv4_dip, dscp = 4, vpnsid = NNI_ANY_PUBLIC_TC2)
+    check_topo_recv_pkt_srh_te(ptfadapter, port=ptf_port_for_PE3_to_P2, dst_ip=test_ipv4_dip, dscp = 4, vpnsid = NNI_ANY_PUBLIC_TC2, segment = SEGMENT_LIST_P2_P1)
+    reset_topo_pkt_counter(ptfadapter)
+```
+
+## Test Plan
+Previous testplan doc "SRv6-phoenixwing-ptf-testplan.md" has detailed the SRv6 test scenarios including SRv6 VPN and SRv6 Policy. In this testplan, We will add Trex traffic check in all these scenarios. The traffic check includes: packet loss, packet encapsulation, traffic ecmp result, and traffic convergence.
+
+### SRv6 VPN
+
+#### SRv6 VPN for single homing
```

Comment:
Can't directly rely on packet counters because of the control packets. Increasing packet size will help. Sonic interface tracks count for different bytes.

Reply:
yes, we will take control packets into account, the control packets is limited, compared to user traffic. so we can reserve some buffer when calculate traffic loss.